### ISSUE TITLE
🐛 Upgrading from `typer<0.12.0` to `typer==0.12.0` breaks the install by partially removing the package/module files

### ISSUE THREAD

**Issue by tiangolo** [tiangolo/typer#790]
This is reproduced and being worked on right now. :nerd_face: 

### Discussed in https://github.com/tiangolo/typer/discussions/784

<div type='discussions-op-text'>

<sup>Originally posted by **brianmedigate** March 31, 2024</sup>
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
$ python3.8 -m virtualenv venv
$ . venv/bin/activate
$ pip install typer==0.9.0
$ pip install typer==0.12.0
$ ll venv/lib/python3.8/site-packages/typer/
total 36
drwxrwxr-x  3 brianm brianm 4096 Mar 31 17:52 ./
drwxrwxr-x 28 brianm brianm 4096 Mar 31 17:52 ../
-rw-rw-r--  1 brianm brianm 9149 Mar 31 17:52 cli.py
-rw-rw-r--  1 brianm brianm 6689 Mar 31 17:52 _completion_classes.py
-rw-rw-r--  1 brianm brianm   30 Mar 31 17:52 __main__.py
drwxrwxr-x  2 brianm brianm 4096 Mar 31 17:52 __pycache__/
```

### Description

After upgrading typer 0.9.0 to 0.12.0 in an existing virtual environment, the `typer` directory in site packages doesn't contain the expected code of typer. It looks like it only contains typer-cli or something like that. I'm guessing it has something to do with the recent packaging changes in https://github.com/tiangolo/typer/pull/780.

### Operating System

Linux

### Operating System Details

Ubuntu 22.04.3

### Typer Version

0.12.0

### Python Version

3.8.18

### Additional Context

_No response_</div>

**Comment by lone17** [comment]
Minor typo in the title: should be `typer<0.12.0` 😄

**Comment by bckohan** [comment]
Yes - I ran into this problem here: https://github.com/bckohan/django-typer/issues/47

I think its a bug in pip. I would expect this to work.

**In my case, since my downstream users did not need the cli, changing the dependency from typer to typer-slim allowed the updates to work.**

**Comment by tiangolo** [comment]
This was fixed in https://github.com/tiangolo/typer/pull/791, released in Typer `0.12.1` :rocket: :tada:

**Comment by washeck** [comment]
Is this really solved?

```
$ pip3 list | grep typer
typer                    0.12.1
$ python3
Python 3.11.6 (main, Nov 30 2023, 18:14:11) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import typer
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'typer'
```
I have typer installed via pipenv as a dependency of `safety`.
